### PR TITLE
Do not check empty strings

### LINKED ISSUE #1354: You cannot specify the same token as both token and nextToken
Got this today : 

`Error running rule maximumNumberOfLines: This is an issue with JSCS and not your codebase.
Please file an issue (with the stack trace below) at: https://github.com/jscs-dev/node-jscs/issues/new`

Error: You cannot specify the same token as both token and nextToken

    at TokenAssert._validateOptions (node_modules/grunt-jscs/node_modules/jscs/lib/token-assert.js:343:15)
    at TokenAssert.linesBetween (node_modules/grunt-jscs/node_modules/jscs/lib/token-assert.js:279:10)
    at Object.module.exports.check (node_modules/grunt-jscs/node_modules/jscs/lib/rules/maximum-number-of-lines.js:38:23)
    at null.<anonymous> (node_modules/grunt-jscs/node_modules/jscs/lib/string-checker.js:141:22)
    at Array.forEach (native)
    at StringChecker._checkJsFile (node_modules/grunt-jscs/node_modules/jscs/lib/string-checker.js:137:31)
    at StringChecker.checkString (node_modules/grunt-jscs/node_modules/jscs/lib/string-checker.js:117:14)
    at null.<anonymous> (node_modules/grunt-jscs/node_modules/jscs/lib/checker.js:43:21)
    at Array.3 (node_modules/grunt-jscs/node_modules/vow/lib/vow.js:699:39)
    at Object.callFns [as _onImmediate] (node_modules/grunt-jscs/node_modules/vow/lib/vow.js:23:35)```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/13894492-you-cannot-specify-the-same-token-as-both-token-and-nexttoken?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by markelog**
Fixes #1354

### CODE REVIEW

**Comment by zxqfox** [comment]
Looks good

File: test/specs/string-checker.js [hunk 1]
```diff
@@ -45,6 +45,15 @@ describe('modules/string-checker', function() {
             });
         });
 
+        it('should not check empty string (#1354)', function() {
+            var spy = sinon.spy(StringChecker.prototype, '_checkJsFile');
```
Comment (mrjoelkemp) [inline test/specs/string-checker.js]: Is there any other way to test this besides spying on a private method?

Comment (zxqfox) [inline test/specs/string-checker.js]: Not sure we should do this ;-( And yep. I don't like this too.

Comment (markelog) [inline test/specs/string-checker.js]: I'm open to suggestions

Comment (mrjoelkemp) [inline test/specs/string-checker.js]: Another not great solution is to assert that none of the rules are called. 

I'm fine with this if you make it protected (we don't have the notion of friend functions) for testing (jsdoc tag change should suffice, comment is a bonus).

**Comment by mrjoelkemp** [comment]
:+1:

**Pull request closed without merging**